internal/graph/preloadedtaskrunner.go

```diff
@@ -0,0 +1,104 @@
+package graph
+
+import (
+	"context"
+	"sync"
+)
+
+// preloadedTaskRunner is a task runner that invokes a series of preloaded tasks,
```

Comment:
Should we be pulling this out into a separate concurrency package?
It might only be useful in the graph now, but not forever!

Reply:
Eventually, yes, but I think for now its okay to keep this in the dispatch package